api/users/controller.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable func-names */
+/* eslint-disable no-confusing-arrow */
+const User = require('./model');
+
+exports.userCreate = function (req, res, next) {
+  const user = new User({
+    email: req.body.email,
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    nickNames: req.body.nickNames,
+    password: req.body.password,
+  });
+  user.save((err) => {
```

Comment:
We also need to be `return`ing promises, so you can change this to:

```javascript
return user.save()
```

and delete the code after this

the route handler will handle proper response sending (check above comment for that code)